src/Eloquent/Model.php

```diff
@@ -216,7 +216,9 @@ public function attributesToArray()
         // Convert dot-notation dates.
         foreach ($this->getDates() as $key) {
```

Comment:
Your change works for `EmbedsOne` but not for `EmbedsMany` since `Arr::has()`/`Arr:set()` doesn't support dot-syntax on nested arrays (`entry.*.date`). Interestingly, `array_get()`/`array_set()` do. So I'd add this to make it work for both `EmbedsOne` and `EmbedsMany`:
```suggestion
        foreach ($this->getDates() as $key) {
            $res = data_get($attributes, $key);
            
            if (is_array($res)) {
                $res = array_filter($res);
            }
            if (Str::contains($key, '.') && Arr::has($attributes, $key)) {
                Arr::set($attributes, $key, $this->serializeDate(
                    $this->asDateTime(Arr::get($attributes, $key))
                ));
            } else if (Str::contains($key, '.') && !empty($res)) {
                data_set($attributes, $key, $this->serializeDate(
                    $this->asDateTime(Arr::get($attributes, $key))
                ));
            }
        }
```

Reply:
@GromNaN Does #3105 support the dot syntax?

Reply:
In fact, the query builder will convert any `DateTimeInterface` into `UTCDatetime`, and inversely for results. Regardless of the path, no need to specify any date casting. That the benefit of having a native date type in MongoDB vs SQL database that store strings for dates.

Reply:
Awesome 👍🏻 Thanks for the clarification